huaweicloud/services/css/resource_huaweicloud_css_scan_task.go

```diff
@@ -0,0 +1,424 @@
+
+	scanTask, err := getScanTaskByName(d, region, conf)
+	if err != nil {
+		common.CheckDeletedDiag(d, err, "CSS cluster scan task")
+	}
+
+	createdAt := utils.PathSearch("create_time", scanTask, 0).(float64) / 1000
+	status := int(utils.PathSearch("status", scanTask, 0).(float64))
+	mErr = multierror.Append(
+		mErr,
+		d.Set("region", region),
+		d.Set("name", utils.PathSearch("name", scanTask, nil)),
+		d.Set("description", utils.PathSearch("desc", scanTask, nil)),
+		d.Set("status", convertClusterScanTaskStatus(status)),
+		d.Set("created_at", utils.FormatTimeStampRFC3339(int64(createdAt), false)),
+		d.Set("smn_status", utils.PathSearch("smn_status", scanTask, nil)),
+		d.Set("smn_fail_reason", utils.PathSearch("smn_fail_reason", scanTask, nil)),
+		d.Set("summary", flattenScanTaskSummaryResponse(scanTask)),
+		d.Set("task_risks", flattenScanTaskRisksResponse(scanTask)),
+	)
+	return diag.FromErr(mErr.ErrorOrNil())
+}
+
+func resourceScanTaskDelete(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	conf := meta.(*config.Config)
+	region := conf.GetRegion(d)
+	clusterID := d.Get("cluster_id").(string)
+
+	// deleteScanTask: delete a scan task.
+	var (
+		deleteScanTaskUrl     = "v1.0/{project_id}/clusters/{cluster_id}/ai-ops/{aiops_id}"
+		deleteScanTaskProduct = "css"
+	)
+
+	deleteScanTaskClient, err := conf.NewServiceClient(deleteScanTaskProduct, region)
+	if err != nil {
+		return diag.Errorf("error creating CSS client: %s", err)
+	}
+	deleteScanTaskPath := deleteScanTaskClient.Endpoint + deleteScanTaskUrl
+	deleteScanTaskPath = strings.ReplaceAll(deleteScanTaskPath, "{project_id}", deleteScanTaskClient.ProjectID)
+	deleteScanTaskPath = strings.ReplaceAll(deleteScanTaskPath, "{cluster_id}", clusterID)
+	deleteScanTaskPath = strings.ReplaceAll(deleteScanTaskPath, "{aiops_id}", d.Id())
+
+	deleteScanTaskOpt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+		MoreHeaders:      map[string]string{"Content-Type": "application/json"},
+	}
+
+	_, err = deleteScanTaskClient.Request("DELETE", deleteScanTaskPath, &deleteScanTaskOpt)
+	if err != nil {
+		return diag.Errorf("error deleting CSS cluster scan task: %s", err)
+	}
+
+	return nil
+}
+
+func buildcreateScanTaskBodyParams(d *schema.ResourceData) map[string]interface{} {
+	bodyParams := map[string]interface{}{
+		"name":        d.Get("name"),
+		"description": utils.ValueIngoreEmpty(d.Get("description")),
+		"alarm":       buildcreateScanTaskAlarmBodyParams(d.Get("alarm")),
+	}
+	return bodyParams
+}
+
+func buildcreateScanTaskAlarmBodyParams(rawParams interface{}) map[string]interface{} {
+	if rawArray, ok := rawParams.([]interface{}); ok && len(rawArray) == 1 {
+		raw := rawArray[0].(map[string]interface{})
+		params := map[string]interface{}{
+			"level":     raw["level"],
+			"smn_topic": raw["smn_topic"],
+		}
+		return params
+	}
+	return nil
+}
+
+func refreshScanTaskID(d *schema.ResourceData, region string, conf *config.Config) error {
```

Comment:
please set the task ID in Create function.
```suggestion
func refreshScanTaskID(d *schema.ResourceData, region string, conf *config.Config) (string, error) {
```